### ISSUE TITLE
Ability to load coordinates for Marks in the F10 map

### ISSUE THREAD

**Issue by iojeda11** [fredemmott/OpenKneeboard#88]
Ability to load coordinates for Marks in the F10 map, in a format that can be changed dynamically from the kneeboard while in DCS

**Comment by bbirchnz** [comment]
I've been looking for a way to do this, but can't currently find any mechanism that works for multiplayer. If anyone knows a technique for getting these out of the lua environment please let me know!

**Comment by iojeda11** [comment]
No idea for multiplayer but this
function MPT.getMarks(_unitName)
	local allMarks =  world.getMarkPanels()
	local _unit = Unit.getByName(_unitName)
	local _playerName = _unit:getPlayerName()
	MPT.target = 0
	MPT.IP = 0

	MPT.wpCount = 0
	MPT.wpList = {}
	if MPT.rowCount(allMarks) > 0 then
		--trigger.action.outText( '2', 1 )
		for i,m in pairs(allMarks) do 
			 if m.author == _playerName then
				
				local tag, value = MPT.parseMark(m.text)
				local k = tonumber(value)
				if tag then
					MPT.wpCount = MPT.wpCount + 1	
					if tag == '+' then
						MPT.target = k--MPT.wpCount
						--trigger.action.outText(MPT.target ,1)
					elseif tag == '>' then
						MPT.IP = k--MPT.wpCount
						--trigger.action.outText(MPT.IP,1)
					end-- if tag == '.t.' then
														
					MPT.wpList[k] = m.pos	
				end
			 end -- if m.autor == _playerName then
		end	--for i,m in pairs(allMarks) do 
	end --if MPT.rowCount(marks) > 0 then
	
	
	return true
end -- function

is how I do it for SP... I suspect you already know that but posting it just in case ...

**Comment by bbirchnz** [comment]
Yep but I don't think that call returns anything when run as a client, and do_stringin('mission') etc doesn't work, as that environment also doesn't exist when you're a client. I've tried dumping all the tables but can't find anything. Please do give it a go though, fresh pair of eyes... If you can find that call I'll find a way to get it into OpenKneeboard

**Comment by iojeda11** [comment]
Ok.. I don't really do MP but I've released some scripts (target tracking)
that have sparked some interest in the MP community, My plan for this
summer is to stand up a server to dive into the MP scripting aspect. I
might accelerate the plan. If it find a way I'll let you know

**Comment by klinkeye** [comment]
There is another project called DCS-Scratchpad which performs this, not sure if its worth seeing how that project is doing it?

**Comment by fredemmott** [comment]
FYI if you’re on the v1.2 betas, you can point a text file tab at the vr scratch pad file, and it will automatically reload when that is modigied

**Comment by iojeda11** [comment]
Excellent! Yes, I'm on the beta branch. Will test over the weekend! Thank
you!

**Comment by fredemmott** [comment]
Since this was originally opened, OpenKneeboard has gained abilities like Window Capture, Web Dashboards, and plugins providing better ways to interact with third-party tools (such as DCS Moving Map, DCS ScratchPad, and DCS Web Editor). It is very unlikely that I will spend significant time on more DCS-specific integration when any developer should be able to do it in ways that integrate with OpenKneeboard, without actually modifying OpenKneeboard.
